src/Microsoft.ApplicationInsights/Extensions/ExceptionExtensions.cs

```diff
@@ -0,0 +1,35 @@
+﻿namespace Microsoft.ApplicationInsights.Extensions
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Globalization;
+
+    internal static class ExceptionExtensions
+    {
+        public static string FlattenMessages(this Exception ex)
+        {
+            var list = new List<string>();
+
+            for (var exWalk = ex; exWalk != null; exWalk = exWalk.InnerException)
+            {
+                list.Add(exWalk.Message);
+            }
+
+            return string.Join(" | ", list);
+        }
+
+        public static string ToLogString(this Exception ex, bool includeStackTrace = false)
+        {
+            string msg = "Type: '{0}' Message: '{1}'";
+
+            var log = string.Format(CultureInfo.InvariantCulture, msg, ex.GetType().ToString(), ex.FlattenMessages());
+
+            if (includeStackTrace)
+            {
+                log = string.Concat(log, "StackTrace: ", ex.StackTrace);
```

Comment:
Do you need Environment.NewLine or something similar to separate it a bit from the messages?

Reply:
I don't think we have any rich text display for ETW messages. 
Could also make an argument is the StackTrace necessary to collect here?
I needed it initially to narrow down the exception last week before i started collecting the InnerExceptions.

What do you think?

Reply:
Since, it's sent to customer IKey, it's typically better to have smaller amount of bytes, while sufficient enough to diagnose the problem. Stack does not seem super-necessary if you show all inner messages already.